sdk/program/src/stake/state.rs

```diff
@@ -16,13 +16,48 @@ use {
 
 pub type StakeActivationStatus = StakeHistoryEntry;
 
+#[repr(u8)]
+#[derive(
+    Debug,
+    Default,
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Clone,
+    Copy,
+    AbiExample,
+    BorshDeserialize,
+    BorshSchema,
+    BorshSerialize,
+)]
+
+pub enum DeactivationFlag {
```

Comment:
I think it's worth spending the time here to slightly generalize `DeactivationFlag` into a set of 8 flag bits for the full additional `u8` we're reclaiming. Then `MustFullyActivateBeforeDeactivationIsPermitted` is just bit 1, and there's 7 more bits that maybe we'll use for something in the future. 

That is, if we're breaking source compat by adding the extra tuple entry to `StakeState::Stake()` then I think it's worth aiming to not break source compat again if we ever want to use more of those 7 remaining bits in the future

Reply:
yeah. sounds good. I will update.